packages/next/server/api-utils/node.ts

```diff
@@ -461,6 +461,7 @@ function setPreviewData<T>(
   data: object | string, // TODO: strict runtime type checking
   options: {
     maxAge?: number
+    path?: string
```

Comment:
I thought it'd be good enough to just accept raw `path` as `string` in this `setPreviewData` function here without extra parsing/validation on it because developers would control the `setPreviewData` call and its parameters, just like they do with redirects inside the `preview` API endpoint. The security concerns around it remain the same - https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.2.4